### PR TITLE
Fix contribute & finalize with callbacks.

### PR DESCRIPTION
**Pull request by gfournieriExec**

### CODE REVIEW

File: test/400_contributeAndCallback.js [hunk 1]
```diff
@@ -574,7 +574,7 @@ contract('Fullchain', async (accounts) => {
                     authorization.taskid, // task      (authorization)
```
Comment (jeremyjams) [inline test/400_contributeAndCallback.js:574]: When do you test `(deal.callback == address(0) && _resultsCallback.length == 0)` path?

Comment (gfournieriExec) [reply test/400_contributeAndCallback.js:576]: acutally we did not test it for now 
image

Comment (jeremyjams) [reply test/400_contributeAndCallback.js:576]: Yes, we should test both cases :)

Comment (gfournieriExec) [reply test/400_contributeAndCallback.js:576]: will check that thanks

Comment (gfournieriExec) [reply test/400_contributeAndCallback.js:576]: The first path is test here : 
https://github.com/iExecBlockchainComputing/PoCo/blob/69f35af63dfd36ec3e07641fd365b57e89c48bcd/test/byContract/IexecPoco/06_claim.test.ts#L288

Comment (gfournieriExec) [reply test/400_contributeAndCallback.js:576]: so actually we are good ?

Comment (jeremyjams) [reply test/400_contributeAndCallback.js:576]: A bit hacky for the path seen in claim test but why not :thinking: 
I'm not sure to understand the other path I see https://github.com/iExecBlockchainComputing/PoCo/blob/69f35af63dfd36ec3e07641fd365b57e89c48bcd/test/400_contributeAndCallback.js#L481 in this test (?)

Comment (gfournieriExec) [reply test/400_contributeAndCallback.js:576]: Thanks for this point, gonna check that

File: test/400_contributeAndCallback.js [hunk 2]
```diff
@@ -600,6 +602,7 @@ contract('Fullchain', async (accounts) => {
                 assert.equal(events[0].args.taskid, taskid);
                 assert.equal(events[0].args.results, web3.utils.utf8ToHex('aResult'));
             });
+            //[TODO] contributeAndFinalize without callback
```
Comment (jeremyjams) [inline test/400_contributeAndCallback.js:605]: :pray:

**Review by jeremyjams — APPROVED** [review]
Thank you :)

**Review by zguesmi — APPROVED** [review]

**Pull request merged**